bin/alpen-client/README.md

```diff
@@ -0,0 +1,628 @@
+│     └─────────────────────────────────────┘                 │
+│                         │                                   │
+│                         ▼                                   │
+│  5. Build ExecBlockPackage (inputs + outputs)               │
+│  6. Create ExecBlockRecord                                  │
+│  7. Store in database                                       │
+│  8. Notify Exec Chain                                       │
+└─────────────────────────────────────────────────────────────┘
+```
+
+**Configuration** ([BlockBuilderConfig](../../../crates/alpen-ee/sequencer/src/block_builder/task.rs)):
+- `blocktime_ms` - Target block interval (default: 1000ms)
+- `max_deposits_per_block` - Deposit throughput limit (default: 16)
+
+---
+
+### Gossip Protocol
+
+**Purpose**: Custom RLPx subprotocol for propagating block headers from sequencer to fullnodes.
+
+**Location**: [crates/reth/node/src/gossip/](../../../crates/reth/node/src/gossip/)
+
+**Responsibilities**:
+- Broadcast new block headers from sequencer
+- Receive and validate block headers from peers
+- Verify sequencer signatures (Schnorr over EIP-191 hash)
+- Forward validated headers to engine control
+
+**Protocol Details**:
+- Protocol name: `alpen_gossip` (version 1)
+- Message signing: Schnorr signatures with sequencer key
+- Verification: All nodes verify against known sequencer pubkey
+
+**Message Structure**:
+```
+AlpenGossipPackage
+├── message: AlpenGossipMessage
+│   ├── header: Block Header
+│   └── seq_no: Sequence Number
+├── public_key: Buf32 (sequencer pubkey)
+└── signature: Buf64 (Schnorr signature)
+```
+
+**Flow**:
+```mermaid
+sequenceDiagram
+    participant Seq as Sequencer
+    participant FN as Fullnode
+
+    Note over Seq: New block produced
+    Note over Seq: sign(header, seq_no)
+    Seq->>FN: AlpenGossipPackage
+    Note over FN: Verify signature
+    Note over FN: Update preconf_rx
+    Note over FN: Trigger block sync
+    FN->>Seq: Reth P2P block request
+```
+
+---
+
+## Data Flows
+
+### Block Sync (Fullnode)
+
+Fullnodes learn about new blocks through two mechanisms:
+
+1. **Consensus (Confirmed/Finalized)**: From OL via OL Tracker
+2. **Pre-confirmed (Latest)**: From sequencer via Gossip Protocol
+
+```mermaid
+sequenceDiagram
+    participant Seq as Sequencer
+    participant FN as Fullnode
+    participant OL as OL Node
+
+    Note over Seq: Produces block N
+
+    Seq->>FN: Gossip: block N header (signed)
+    FN->>FN: Verify signature
+    FN->>FN: Update preconf head
+    FN->>Seq: Reth P2P: request block N
+    Seq-->>FN: Block N data
+    FN->>FN: Execute & validate
+
+    Note over OL: Block N confirmed on OL
+
+    OL->>FN: OL Tracker: consensus update
+    FN->>FN: Mark block N as safe
+```
+
+**Important**: EE fullnodes do not have independent block sync. They rely on:
+- Reth's P2P for block data (triggered by gossip)
+- OL for consensus (confirmed/finalized status)
```

Comment:
this is not quite block sync, right? (not only that technically this does not sync blocks, but) we are not syncing data from OL, it is needed to be fetched from bitcoin. This is not implemented right now, correct?